### ISSUE TITLE
Memory leak in manager pod

### ISSUE THREAD

**Issue by jlkaus** [fybrik/fybrik#1409]
I'm seeing a slow memory leak in the manager pod as I perform many iterations of the following sequence:
- create fybrikapplication with no assets
- add an WKC connected asset to the application
- Use flight to access the data in the asset
- remove the asset from the application
- remove the application

I've run this thousands of times.  Eventually the manager pod consumes all the available node memory (~55GB in my case) and the node OOM killer terminates the container, causing the manager pod to restart.  My testing resumes and eventually it runs out of memory again.

Originally, we had a memory limit on the manager pod of 1GB, which of course causes the pod to be restarted much much quicker due to this leak.

This was seen in v0.6.3 and v0.6.4.  Maybe have existed before that. I wasn't looking as close at manager restarts.
@laurieaw63

**Comment by jlkaus** [comment]
Just to put some more specific numbers on this, I restarted the manager pod, to get rid of any leaked memory, and the size of the pod after restart was 20Mi.

After my 1000 iterations, it was 31984Mi.  So, roughly 32Mi leaked per fybrikapplication/asset/use/remove cycle.

**Comment by roytman** [comment]
I run in a loop the following script to reproduce the issue. 
```bash
#!/usr/bin/env bash

echo "Create empty fybrikapplication"
kubectl  apply -f fappE.yaml
while [[ $(kubectl get fybrikapplication my-notebook -n fybrik-notebook-sample -n fybrik-notebook-sample -o 'jsonpath={.status.ready}') != "true" ]]; do echo "waiting for FybrikApplication" 
&& sleep 5; done

echo "Add asset"
kubectl  apply -f fapp.yaml
while [[ $(kubectl get fybrikapplication my-notebook -n fybrik-notebook-sample -o 'jsonpath={.status.assetStates.fybrik-notebook-sample/paysim-csv.conditions[?(@.type == "Ready")].status}') 
!= "True" ]]; do echo "waiting for fybrik-notebook-sample/paysim-csv asset" && sleep 5; done

echo "Remove asset"
kubectl  apply -f fappE.yaml

while [[ `kubectl get pods -n fybrik-blueprints | grep my-notebook-fybrik-notebook-sample-arrow-flight | wc -l` != 0 ]]; do echo "waiting for moduel pod " && sleep 5; done

kubectl delete fybrikapplications.app.fybrik.io my-notebook -n fybrik-notebook-sample
```

I did not do it for 1000 iterations, just for 100. Without the fix from #1414, the manager pod memory was increased from 31Mi to 1604Mi. After the fix, the memory footprint was 60Mi.
[fapp.yaml.txt](https://github.com/fybrik/fybrik/files/8430101/fapp.yaml.txt)
[fappE.yaml.txt](https://github.com/fybrik/fybrik/files/8430102/fappE.yaml.txt)

**Comment by roytman** [comment]
memory heap profiles before the fix
at start 
![origStart](https://user-images.githubusercontent.com/1816654/162059384-3071501f-5495-4fc9-9af4-a462e8204724.svg)
after the 100 iterations
![origFinal](https://user-images.githubusercontent.com/1816654/162059406-81524c97-1895-4137-9a8a-407b49a13931.svg)
 
after the fix and after running the 100 iterations.
![fixedFinal](https://user-images.githubusercontent.com/1816654/162059574-be47b57d-240d-4927-8ac8-ef7c82fcf96b.svg)
